pkg/controller/signup.go

```diff
@@ -60,3 +62,38 @@ func (s *Signup) GetHandler(ctx *gin.Context) {
 		ctx.JSON(http.StatusOK, signupResource)
 	}
 }
+
+// VerifyCodeHandler validates the phone verification code passed in by the user
+func (s *Signup) VerifyCodeHandler(ctx *gin.Context) {
+	code := ctx.Param("code")
```

Comment:
if this code is empty, maybe we should just quit instead of moving ahead to retrieve the usersignup?

Reply:
as mentioned in the other comment, maybe code actually can't be empty here since it's a path parameter... if it's empty the endpoint probably wasn't hit..?